Calling in operator for Array of Sum Type using one of the types mentioned in the Sum Type is unstable and may cause compilation error

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
OS: linux, Debian GNU/Linux 10 (buster) (WSL 2)
Processor: 12 cpus, 64bit, little endian, AMD Ryzen 5 4600H with Radeon Graphics
CC version: cc (Debian 8.3.0-6) 8.3.0

getwd: /home/work/v
vmodules: /home/work/.vmodules
vroot: /home/work/v
vexe: /home/work/v/v
vexe mtime: 2021-10-20 11:39:53
is vroot writable: true
is vmodules writable: true
V full version: V 0.2.4 7c1fff3

Git version: git version 2.20.1
Git vroot status: weekly.2021.42-19-g7c1fff34
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 333c14de

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg bug.v`
```v
type StringOrNumber = int | string

fn main() {
  arr := [StringOrNumber(1), 2, 'test']
  println(1 in arr)
}

```

**What did you expect to see?**

Depending on implementation: `true` or `false` or assertion error comparing wrong types


**What did you see instead?**
```
==================
/tmp/v/bug.4789833462941591359.tmp.c:10610: error: cannot convert 'int' to 'struct main__StringOrNumber'
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

I also noticed V playground can't process:
```v
type StringOrNumber = int | string
arr := [StringOrNumber(1), 2, 'test']
println(StringOrNumber(1) in arr)
```

or

```v
type StringOrNumber = int | string
arr := [StringOrNumber(1), 2, 'test']
t := arr[0]
println(t in arr)
```
It results in `Time limit exceeded (wall clock)`

These two examples work on my machine though.

One more observation:
On my machine when calling via a string the program doesn't return anything back. E.g.

```v

type StringOrNumber = int | string

fn main() {
  arr := [StringOrNumber(1), 2, 'test']
  println('a' in arr)
}
```
Doesn't provide any output.

--- a/vlib/v/gen/c/infix.v
+++ b/vlib/v/gen/c/infix.v
@@ -437,12 +437,12 @@ fn (mut g Gen) infix_expr_in_op(node ast.InfixExpr) {
 			elem_type := right.sym.info.elem_type
 			elem_type_ := g.unwrap(elem_type)
 			if elem_type_.sym.kind == .sum_type {
-				if node.left_type in elem_type_.sym.sumtype_info().variants {
+				if ast.mktyp(node.left_type) in elem_type_.sym.sumtype_info().variants {
 					new_node_left := ast.CastExpr{
 						arg: ast.empty_expr
 						typ: elem_type
 						expr: node.left
-						expr_type: node.left_type
+						expr_type: ast.mktyp(node.left_type)
 					}
 					g.gen_array_contains(node.right_type, node.right, new_node_left)
 					return
@@ -507,12 +507,12 @@ fn (mut g Gen) infix_expr_in_op(node ast.InfixExpr) {
 			elem_type := right.sym.info.elem_type
 			elem_type_ := g.unwrap(elem_type)
 			if elem_type_.sym.kind == .sum_type {
-				if node.left_type in elem_type_.sym.sumtype_info().variants {
+				if ast.mktyp(node.left_type) in elem_type_.sym.sumtype_info().variants {
 					new_node_left := ast.CastExpr{
 						arg: ast.empty_expr
 						typ: elem_type
 						expr: node.left
-						expr_type: node.left_type
+						expr_type: ast.mktyp(node.left_type)
 					}
 					g.gen_array_contains(node.right_type, node.right, new_node_left)
 					return

--- a/vlib/v/tests/in_expression_test.v
+++ b/vlib/v/tests/in_expression_test.v
@@ -319,3 +319,11 @@ fn test_in_array_literal_of_sumtype() {
 	val2 := `+`
 	assert val2 in [TokenValue(`+`), TokenValue(`-`)]
 }
+
+type StringOrNumber = int | string
+
+fn test_in_array_of_sumtype() {
+	arr := [StringOrNumber(1), 2, 'test']
+	println(1 in arr)
+	assert 1 in arr
+}

